### ISSUE TITLE
error : cannot resolve `std::borrow::Cow<'_, str>: std::borrow::Borrow<_>

### ISSUE THREAD

**Issue by vamshi9666** [SergioBenitez/Rocket#643]
rocket version : 0.3.10
rustc version :  1.28.0-nightly
i am getting started into rocket .
i have installation as in docs and  downloaded examples from git .
when i entered cargo run .
i see
```
 type annotations required: cannot resolve `std::borrow::Cow<'_, str>: std::borrow::Borrow<_>`
   --> lib/src/http/header.rs:544:47
    |
544 |                 Header::new(key.as_str(), val.borrow())
    |                                               ^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0283`.
error: Could not compile `rocket`.
i tied to switch to other versions and i faced more error in previous versions.
if this issue is already in progress . please provide link or reference 

```

**Comment by logual** [comment]
Hi Sergio,

The same problem :-(

stable-x86_64-apple-darwin unchanged - rustc 1.26.0 (a77568041 2018-05-07)
nightly-x86_64-apple-darwin unchanged - rustc 1.28.0-nightly (952f344cd 2018-05-18)

rustup override set nightly
rustup update && cargo update

But no results:

<img width="768" alt="screen shot 2018-05-19 at 14 31 19" src="https://user-images.githubusercontent.com/1644902/40267655-777ebeee-5b71-11e8-9577-e1f1fc4009f6.png">

**Comment by CallistoM** [comment]
Having the same problem, tried to downgrade to version 0.3.6, cargo does not like this. 

Running the following nightly version:
- cargo 1.28.0-nightly (f352115d5 2018-05-15)
- rustc 1.28.0-nightly (952f344cd 2018-05-18)

![capture](https://user-images.githubusercontent.com/11871820/40268383-941e1efa-5b6c-11e8-92a5-70b20cd0283a.PNG)

**Comment by alex1702** [comment]
my temporary solution is to downgrade:
```bash
rustup toolchain install nightly-2018-05-17
rustup default nightly-2018-05-17
```

**Comment by vamshi9666** [comment]
is this resolved ?

**Comment by bjorn3** [comment]
Not yet #644 isn't merged yet

**Comment by vamshi9666** [comment]
did downgrading worked for anyone ?

**Comment by logual** [comment]
Downgrading fixed the problem

**Comment by SergioBenitez** [comment]
Fixed in ff5db841114d8da1a17b723808501c756df57c64. Will push a new release with this fix soon.
